### PR TITLE
Add Consul/Nomad based service-discovery

### PR DESCRIPTION
**Pull request by sportebois**
This change request extends the current Docker startup script for VerneMQ and adds support for Consul-registered Nomad services.

### Changes

- Dockerfile container the new environment variable for Consul based auto clustering
- README.md has been updated to document the setup
- the startup bash script now call Consul to look for other nodes of an existing cluster, and join it if such service is found

### Out of scope

Because of time constraint, this isn't 100% completed, and works well when the Nomad task driver is set to run with host networking mode. 
With such configuration, the update has been tested to and auto-joining worked as expected both for cold start (when multiple tasks from the same job start for the first time, simultaneously), and for rolling updates. 
Making this work in bridge mode with dynamic ports should ideally be done too, be will be part of another PR in the future.

### CODE REVIEW

**Review by julienlevasseur — APPROVED** [review]
GTG :+1:

File: README.md [hunk 1]
```diff
@@ -94,6 +94,80 @@ If you started VerneMQ cluster inside Kubernetes using ```DOCKER_VERNEMQ_DISCOVE
...
+
+```hcl
+job "vernemq" {
+  type = "service"
+
+  group "mqtt" {
+    task "mqtt" {
+      driver = "docker"
+      config {
+        # Vecause of the autodiscovery, name and IP, we use host networking (we get the same kidn of routing than clusterIP in Kubernetes)
```
Comment (kritchie) [inline README.md]: Little typos over here 
`V/B/ecause`
`ki/dn/nd`

**Review by kritchie — APPROVED** [review]
I spotted 2 typos. The rest LGTM !

File: README.md (hunk 1 above)
Comment (sportebois) [reply README.md]: Thanks, addressed in 82e9ca43e705b71e24700ff2e9a69f9968f81cea (with some weird spacing to in the example, leftovers from the removal of other non-relevant parts of the configuration....)

**Pull request merged**